lib/OptimizationManopt/test/runtests.jl

```diff
@@ -105,6 +105,89 @@ end
     @test sol.minimum < 0.1
 end
 
+@testset "CMA-ES" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.CMAESOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock)
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 0.1
+end
+
+@testset "ConvexBundle" begin
+    using QuadraticModels, RipQP
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.ConvexBundleOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock, AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt, sub_problem= Manopt.convex_bundle_method_subsolver!) 
+    @test sol.minimum < 0.1
+end
+
+@testset "TruncatedConjugateGradientDescent" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.TruncatedConjugateGradientDescentOptimizer()
+
+    optprob = OptimizationFunction(rosenbrock, AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test_broken sol.minimum < 0.1
```

Comment:
It gives NaNs as the minimizer after solving smh, even though printing the optimization variables shows non-NaN values the returned answer from Manopt is

Reply:
I do not understand much of the tests you are running, its a bit sparse on explanations and hence reads for me like an unknown language.

But be aware that TCG is an algorithm that only solves problems on a tangent space (namely it is the default subsolver for trust regions). So if you run it on a manifold, that might lead to a lot of unexpected behaviour. And yes, when we allow manifolds therein that might for sure be considered a bug, but I have not yet checked that.

Reply:
Yes you should mention it in the docs and/or throw an error

Reply:
It would work on Euclidean as well; but I am surprised this now got merged with TCG still in here. The docs also clearly state that this just solves a problem in the tangent space of the given iterate (so basically the initial X is the iterate https://manoptjl.org/stable/solvers/truncated_conjugate_gradient_descent/

Reply:
It's not still there https://github.com/SciML/Optimization.jl/pull/763/commits/6af1a295cff23c1e97d69f27afcf2f3090218acb

Ah, my main reference was https://manoptjl.org/stable/solvers/ s it wasn't obvious that it is distinct from the other methods in that list (you are right it is mentioned in the text)

Reply:
I can highlight a few words in boldface from that list: 

* ``[Steihaug-Toint Truncated Conjugate-Gradient Method](https://manoptjl.org/stable/solvers/truncated_conjugate_gradient_descent/#tCG) a solver for a constrained problem defined ->>>on a tangent space<<<<-`` should we add rotating emojis around that? I am not sure how to improve that.

Reply:
Since at the top of the page it says argmin_{p \in M } f(p) and then goes on to list methods that can be used to solve such problems, it isn't obvious that any of methods in the list that are limited to p \in TM